src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -11,58 +11,59 @@ import com.github.quillraven.fleks.collection.compareEntity
 data class Entity(val id: Int)
 
 /**
- * DSL marker for the three different entity contexts: hook, create and update.
- */
-@DslMarker
-annotation class EntityCtxMarker
-
-/**
- * A DSL class for basic [Entity] extension functions within an add/remove hook of a [Component] or [Family].
- * These extensions are also used by [World.query] and [compareEntity].
- *
- *
- * Also, the same functionality is provided by a [Family] and an [IteratingSystem] but not via this class.
- * The methods must be manually copied in all those areas because unfortunately, I have no better idea how to solve that :(
+ * A class for basic [Entity] extension functions within an add/remove hook of a [Component], [Family],
+ * [IntervalSystem], [World] or [compareEntity].
  */
-@EntityCtxMarker
-open class EntityHookContext(
+abstract class BaseEntityExtensions(
```

Comment:
Consider not using plural form for class names. Describe the role of the provided extensions instead, e.g. `EntityHandler`, `ComponentContext`.

Reply:
I already have an `EntityCreateContext` and `EntityUpdateContext`. What do you think about calling it `EntityBaseContext` or `EntityBaseComponentContext`?

Reply:
I already have an `EntityCreateContext` and `EntityUpdateContext`. What do you think about calling it `EntityBaseContext` or `EntityBaseComponentContext`?